test/extensions/transport_sockets/tls/test_private_key_method_provider.cc

```diff
@@ -119,7 +120,17 @@ static ssl_private_key_result_t rsaPrivateKeySign(SSL* ssl, uint8_t* out, size_t
     return ssl_private_key_failure;
   }
 
-  if (!calculateDigest(md, in, in_len, hash, &hash_len)) {
+  in2.assign(in, in + in_len);
+
+  // If crypto error is set, we'll modify the incoming token by flipping
+  // the bits.
+  if (ops->test_options_.crypto_error_) {
+    for (size_t i = 0; i < in_len; i++) {
+      in2.data()[i] ^= in2.data()[i];
```

Comment:
no need `.data()` (`in2[i]` should just work), and this is just clearing in2 to zero? x^x is always 0...

Reply:
No, XOR was supposed to be NOT -- I was trying to flip the bits in the token but managed to think it wrong. :-) I'll make a fix.